clients/android/app/src/main/java/app/lockbook/InitialLaunchFigureOuter.kt

```diff
@@ -1,25 +1,127 @@
 package app.lockbook
 
-import android.app.Activity
+import android.content.Context
 import android.content.Intent
+import android.content.SharedPreferences
 import android.os.Bundle
-import app.lockbook.core.isDbPresent
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.biometric.BiometricConstants
+import androidx.biometric.BiometricManager
+import androidx.biometric.BiometricPrompt
+import androidx.core.content.ContextCompat
 import app.lockbook.core.loadLockbookCore
+import app.lockbook.loggedin.mainscreen.MainScreenActivity
+import app.lockbook.login.WelcomeActivity
+import app.lockbook.utils.SharedPreferences.BIOMETRIC_NONE
+import app.lockbook.utils.SharedPreferences.BIOMETRIC_OPTION_KEY
+import app.lockbook.utils.SharedPreferences.BIOMETRIC_RECOMMENDED
+import app.lockbook.utils.SharedPreferences.BIOMETRIC_STRICT
+import app.lockbook.utils.SharedPreferences.LOGGED_IN_KEY
+import app.lockbook.utils.SharedPreferences.SHARED_PREF_FILE
 
-
-class InitialLaunchFigureOuter : Activity() {
+class InitialLaunchFigureOuter : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        setContentView(R.layout.splash_screen)
         loadLockbookCore()
 
-        if (isDbPresent(filesDir.absolutePath)) {
-            val intent = Intent(this, ListFiles::class.java)
-            startActivity(intent)
-            finish()
+        val pref = getSharedPreferences(SHARED_PREF_FILE, Context.MODE_PRIVATE)
+
+        if (pref.getBoolean(LOGGED_IN_KEY, false)) {
+            if (!isBiometricsOptionsAvailable() && pref.getInt(
+                BIOMETRIC_OPTION_KEY,
+                BIOMETRIC_NONE
+            ) != BIOMETRIC_NONE
+            ) {
+                pref.edit()
+                    .putInt(BIOMETRIC_SERVICE, BIOMETRIC_NONE)
+                    .apply()
+            }
+            checkBiometricOptions(pref)
         } else {
-            val intent = Intent(this, Welcome::class.java)
+            val intent = Intent(this, WelcomeActivity::class.java)
             startActivity(intent)
             finish()
         }
     }
-}
\ No newline at end of file
+
+    private fun isBiometricsOptionsAvailable(): Boolean =
+        BiometricManager.from(applicationContext)
+            .canAuthenticate() == BiometricManager.BIOMETRIC_SUCCESS
+
+    private fun checkBiometricOptions(pref: SharedPreferences) {
```

Comment:
a name like `checkBiometricOptions` is very similar to `isBiometricsOptionsAvailable`. You probably want to call this something like `performBiometricFlow`